src/game/logic/scriptengine/scriptengine.cpp

```diff
@@ -13,56 +13,9401 @@
+    m_actionTemplates[ScriptAction::SOUND_AMBIENT_RESUME].m_uiStrings[0] = "Resume the ambient sounds.";
+
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_internalName = "MUSIC_SET_TRACK";
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_uiName = "Multimedia_/Music/Play a music track.";
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_numParameters = 3;
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_parameters[0] = Parameter::MUSIC;
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_parameters[1] = Parameter::BOOLEAN;
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_parameters[2] = Parameter::BOOLEAN;
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_numUiStrings = 4;
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_uiStrings[0] = "Play ";
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_uiStrings[1] = " using fadeout (";
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_uiStrings[2] = ") and fadein (";
+    m_actionTemplates[ScriptAction::MUSIC_SET_TRACK].m_uiStrings[3] = ").";
+
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_BEGIN].m_internalName = "CAMERA_LETTERBOX_BEGIN";
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_BEGIN].m_uiName = "Camera_/Letterbox/Start letterbox mode.";
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_BEGIN].m_numParameters = 0;
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_BEGIN].m_numUiStrings = 1;
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_BEGIN].m_uiStrings[0] = "Start letterbox mode (hide UI, add border).";
+
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_END].m_internalName = "CAMERA_LETTERBOX_END";
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_END].m_uiName = "Camera_/ End letterbox mode.";
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_END].m_numParameters = 0;
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_END].m_numUiStrings = 1;
+    m_actionTemplates[ScriptAction::CAMERA_LETTERBOX_END].m_uiStrings[0] = "End letterbox mode (show UI, remove border).";
+
+    m_actionTemplates[ScriptAction::CAMERA_BW_MODE_BEGIN].m_internalName = "CAMERA_BW_MODE_BEGIN";
+    m_actionTemplates[ScriptAction::CAMERA_BW_MODE_BEGIN].m_uiName = "Camera_/Fade Effects/Start black & white mode.";
+    m_actionTemplates[ScriptAction::CAMERA_BW_MODE_BEGIN].m_numParameters = 1;
+    m_actionTemplates[ScriptAction::CAMERA_BW_MODE_BEGIN].m_parameters[0] = Parameter::INT;
+    m_actionTemplates[ScriptAction::CAMERA_BW_MODE_BEGIN].m_numUiStrings = 1;
+    m_actionTemplates[ScriptAction::CAMERA_BW_MODE_BEGIN].m_uiStrings[0] = "Frames to fade into black & white mode = ";
+
+    m_actionTemplates[ScriptAction::CAMERA_BW_MODE_END].m_internalName = "CAMERA_BW_MODE_END";
+    m_actionTemplates[ScriptAction::CAMERA_BW_MODE_END].m_uiName = "Camera_/Fade Effects/End black & white mode.";
+    m_actionTemplates[ScriptAction::CAMERA_BW_MODE_END].m_numParameters = 1;
```

Comment:
number of parameters does not match set parameters.

Reply:
Original code does this.